packages/examples/usage-tracking/usage-tracking.js

```diff
@@ -0,0 +1,106 @@
+// Import the Fixie client
+import { FixieClient } from 'fixie';
+import 'dotenv/config';
+import axios from 'axios';
+import { encode } from 'gpt-3-encoder';
+
+// Set the API key and Create the Fixie client
+const FIXIE_API_KEY = process.env.FIXIE_API_KEY;
+
+// Set our Agent ID that we want to get usage info on
+const AGENT_ID = process.env.FIXIE_AGENT_ID;
+
+// const fixieClient = new FixieClient({ apiKey: FIXIE_API_KEY });  // TODO once we have the right method
+
+let numConversations = 0;
+let numMessages = 0;
+let lenConversations = 0;
+let numTokens = 0;
+
+console.log('calling listAgentConversations');
+getAgentConversations(AGENT_ID).then((data) => {
+  numConversations = data.length;
+
+  console.log(`Got back ${data.length} conversations`);
+  data.forEach((element) => {
+    console.log('------------------------------------------------------------\n');
+    console.log(`Conversation ID: ${element.id}`);
+    console.log(`Total conversation turns\t${element.turns.length}`);
+
+    getConvoTurns(element);
+
+    // We want to get content for any messages that are:
+    // conversation.role = "user" -> get messages that are messages.kind = "text" and get messages.content for the text of the message
+    //      make sure the state is "done"
+    // conversation.role = "assistant"
+    //      kind = "text", get the content
+    //      make sure the state is "done"
+    //
+    //      kind = "functionResponse"....get the response
+  });
+
+  console.log('\n\n============================================================');
+  console.log(`Final stats for agent ${AGENT_ID}:\n`);
+  console.log(`Total Conversations\t${numConversations}\n`);
+  console.log(`Total Agent Messages\t${numMessages}\n`);
+  console.log(`Total Characters\t${lenConversations}\n`);
+  console.log(`Total LLM Tokens\t${numTokens}\n`);
+  console.log('============================================================');
+});
+
+function getConvoTurns(conversation) {
+  let numConvoMessages = 0;
+  let numConvoChars = 0;
+  let numConvoTokens = 0;
+
+  // Iterate thru the conversation and process all the turns
+  conversation.turns.forEach((turn) => {
+    numConvoMessages += turn.messages.length;
+
+    // Iterate through the turn messages and log their stats
+    turn.messages.forEach((message) => {
+      // console.log(message);
+      numMessages++;
+      // Make sure it's a message type that we want
+      if (turn.state == 'done') {
+        // Function Response
+        if (turn.role == 'assistant' && message.kind == 'functionResponse') {
+          lenConversations += message.response.length;
+          const encoded = encode(message.response);
```

Comment:
I don't know for sure that this is the right encoder, it might be though. The one we use in Python land is 

```
import tiktoken

# OpenAI models that we care about (GPT-3.5, GPT-4, text-embedding-ada-002) use cl100k_base.
#   https://github.com/openai/tiktoken/blob/main/tiktoken/model.py
#   https://github.com/openai/openai-cookbook/blob/main/examples/How_to_count_tokens_with_tiktoken.ipynb
encoder = tiktoken.get_encoding("cl100k_base")
```

Reply:
i found a js version of tiktoken (js-tiktoken) and have moved to that so we get consistent numbers across the js and python versions of this script.